client/src/controllers/InitController.ts

```diff
@@ -41,15 +62,30 @@ export class InitController extends Controller<HTMLElement> {
    * Support the ability to also dispatch custom event names.
    */
   ready() {
-    const events = this.eventValue.split(' ').filter(Boolean);
     const delayValue = this.delayValue;
+    const detail = { ...(this.detailValue || {}) }; // Coerce to object
+    const events = this.eventValue.split(' ').filter(Boolean);
 
     debounce(() => true, delayValue < 0 ? null : delayValue)().then(() => {
       this.element.classList.add(...this.readyClasses);
       this.element.classList.remove(...this.removeClasses);
-      this.dispatch('ready', { bubbles: true, cancelable: false });
+
+      const readyEvent = this.dispatch('ready', {
+        bubbles: true,
+        cancelable: false,
+        detail,
+      });
```

Comment:
Note that you still have `cancelable: false` here, this will mean that calling event.preventDefault() on the event will throw an error. Instead, here's the same code but with cancelable back to the default (which is true).

```suggestion
const readyEvent = this.dispatch('ready', { bubbles: true, detail });

```

Alternatively, we can be explicit here.

```suggestion
      const readyEvent = this.dispatch('ready', {
        bubbles: true,
        cancelable: true,
        detail,
      });
      
```

Reply:
I think I'm in favour of being explicit. The next person might not be aware of the default behaviour